ginga/rv/plugins/LineProfile.py

```diff
@@ -275,45 +303,55 @@ def _slice(self, naxes, mk):
 
         # For axis > 3
         for i in range(2, naxes):
-            slice_obj[i] = int(round(self.image.revnaxis[i-2] + 1))
+            if i != i_sel:
+                slice_obj[i] = 0  # int(round(self.image.revnaxis[i-2] + 1))
 
         # Slice selected axis
-        slice_obj[self.selected_axis-1] = slice(None, None, None)
+        slice_obj[i_sel] = slice(None, None, None)
 
         return slice_obj
 
     def get_axis(self, i):
         try:
-            self.x_lbl = self.image.get_keyword('CTYPE%d' % i, None)
-            try:
-                kwds = ['CRVAL%d' % i, 'NAXIS%d' % i, 'CDELT%d' % i]
-                crval_i, naxis_i, cdelt_i = self.image.get_keywords_list(*kwds)
+            naxis_s = 'NAXIS{}'.format(i)
+            naxis_i = self.image.get_keyword(naxis_s)
+            self.x_lbl = self.image.get_keyword('CTYPE{}'.format(i), naxis_s)
 
+            try:
+                kwds = ['CRVAL{}'.format(i), 'CDELT{}'.format(i)]
+                crval_i, cdelt_i = self.image.get_keywords_list(*kwds)
             except KeyError as e:
-                raise ValueError("Missing FITS keyword: %s" % str(e))
-
-            axis = crval_i + np.arange(0, naxis_i, 1) * cdelt_i
-
-            if self.x_lbl is not None:
-                units = self.image.get_keyword('CUNIT%d' % i, None)
-                if units is not None:
-                    self.x_lbl += (' (%s)' % str(units))
-            else:
-                self.x_lbl = ''
-            # Assume Y label should always be flux?
-            self.y_lbl = 'Flux'
-            return axis
+                self.logger.error("Missing FITS keyword: {}".format(str(e)))
+                crval_i = 0
+                cdelt_i = 1
+
+            axis = crval_i + np.arange(naxis_i) * cdelt_i
+            self._crval = crval_i
+            self._cdelt = cdelt_i
+
+            units = self.image.get_keyword('CUNIT{}'.format(i), None)
+            if units is not None:
+                self.x_lbl += (' ({})'.format(units))
+
+            # Get pixel value info from header
+            self.y_lbl = self.image.get_keyword('BTYPE', 'Flux')
+            bunit = self.image.get_keyword('BUNIT', None)
```

Comment:
I find that calibrated HST images and also the example image from SunPy has these keywords, so using these to set Y-axis label.